### PR TITLE
Added OTP screen and modified other files

### PR DESCRIPTION
**Pull request by Rados1993**
Added otp_screen, and modified register_screen and welcome_screen.

### CODE REVIEW

File: lib/screens/welcome_screen.dart [hunk 1]
```diff
@@ -58,7 +58,7 @@ class _WelcomeScreenState extends State<WelcomeScreen> {
                   height: 50,
                   child: CustomButton(
                     onPressed: () {
-                      if (ap.isSignedIn == false) {
+                      if (ap.isSignedIn == true) {
```
Comment (lecicdjuro) [inline lib/screens/welcome_screen.dart]: you  don't need to check is true == true
it it's enough if(ap.isSignedIn){
}

**Review by lecicdjuro — CHANGES_REQUESTED** [review]

File: lib/screens/otp_screeen.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                          ),
+                          const SizedBox(height: 20),
+                        ],
+                    ),
+                ),
+            ),
+        ),
+    );
+  }
+}
```
Comment (lecicdjuro) [inline lib/screens/otp_screeen.dart]: format this class

File: lib/utils/utils.dart [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+void showSnackBar(BuildContext context, String content) {
```
Comment (lecicdjuro) [inline lib/utils/utils.dart]: You can create some class ScreenUtils that have static methods like this one.

Comment (lecicdjuro) [reply lib/utils/utils.dart]: and later you can use it on this way, example ScreenUtils.showSnackBar(context, 'Error happened')

File: lib/utils/utils.dart [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
-void showSnackBar(BuildContext context, String content) {
-  ScaffoldMessenger.of(context).showSnackBar(
+class ScreenUtils {
+  static void showSnackBar(BuildContext context, String content) {
+    ScaffoldMessenger.of(context).showSnackBar(
```
Comment (lecicdjuro) [inline lib/utils/utils.dart:6]: where are you calling this method?

**Review by lecicdjuro — CHANGES_REQUESTED** [review]

File: lib/utils/utils.dart (hunk 4 above)
Comment (Rados1993) [reply lib/utils/utils.dart:6]: import 'package:firebase_auth/firebase_auth.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:phoneauth/screens/otp_screeen.dart';
import 'package:phoneauth/utils/utils.dart';
import 'package:shared_preferences/shared_preferences.dart';

class AuthProvider extends ChangeNotifier {
  bool _isSignedIn = false;
  bool get isSignedIn => _isSignedIn;

  final FirebaseAuth _firebaseAuth = FirebaseAuth.instance;

  AuthProvider() {
    checkSign();
  }

  void checkSign() async {
    final SharedPreferences s = await SharedPreferences.getInstance();
    _isSignedIn = s.getBool("is_signedin") ?? false;
    notifyListeners();
  }

  void signInWithPhone(BuildContext context, String phoneNumber) async {
    try {
      await _firebaseAuth.verifyPhoneNumber(
        phoneNumber: phoneNumber,
          verificationCompleted: (PhoneAuthCredential phoneAuthCredential) async {
            await _firebaseAuth.signInWithCredential(phoneAuthCredential);
          },
          verificationFailed: (error) {
            throw Exception(error.message);
      },
          codeSent: (verificationId, forceResendingToken) {
            Navigator.push(context, MaterialPageRoute(builder: (context)=> OtpScreen(verificationId: verificationId),),);
          },
          codeAutoRetrievalTimeout: (verificationId) {});
    } on FirebaseAuthException catch (e) {
      showSnackBar(context, e.message.toString());
    }
  }
}
Nije zavrsena strana auth_provider

**Review by lecicdjuro — APPROVED** [review]

**Pull request merged**